### PR TITLE
feat(useAnimate): new function

### LINKED ISSUE #2077: Add a new function useAnimate
### Clear and concise description of the problem

Controllable and high-performance animation based on [Web Animations API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Animations_API).

While there are many ways to implement animations, the advantage of this approach is that:

- Easy to use, only need to define keyframes for animation
- Ability to get and control animation play status through js
- ...

Of course I can try to accomplish this.

### Suggested solution

[Basic demo](https://play.vueuse.org/#N4IgDghgxg1hDmBTAziAXAbVAOwgW0XRADcBXRAWgBNE8BLEAGhGQHtSAnKQtEU7MDHhMQNZFA50wAFzqtsRAGrkABABFadFQAoAFhAA2AMxV1sKgGIdE2KLoCUp5CogqaxRAdZgz8FaVkDOmkATxcDLwB3ZxD2FWlWFUjJaUQVAFVsOg8OZEMVZTSAGToAIw4ISRQVI1YOAtUAJhUAMhUAZhFOAyJdaWkwZDQAemH+QXgAOihWPGGyShp6YaDS4bMaAA9JvAArVABfRhx8HhAAAQXSZERh5H1rKhE2Tm4icaERMQkpWXkiADKD0QVAaiHSN38gWCdBQky6HB6vD6AyGow+Uxmc0u5Gut3ulRB5wAnJMAAzk9bYLY7fYgI4nAhEHGIPHDGbWZ7sLhnDFfFA-GRyBS8ADCrAiiCgf3MrBMKBu2Fk+UKKnFeDArGQMPkGWhshQCKRIBRgxGYwEQmms2GLLZHMQJPJlI2iG2e0OAF1mEY6AZDWhQABBMBgSYLdCgVIagwQVJEAA8VGyAB1zCpNcgALwpkDWWOyDy5tMqGr+zY5kAAP2L6d211kRhCle4SsQHFrpcMdHg2ErwVoyAordSHZAJaSlfE1hsnZUuinEkQs-H6dYOSMUUrujoVBo2FrAD4Jwm6Hh4BPS9YjJXPHPS5FK+0yWSwJt7-Ony+3x-kFxt-0ZqjBUkSTPAwS6KQpR4lw8ipEq1pzFcNzzLiKF4BAZjDJAsAICg2FQUEUArKw8CsBQOSyFAhjUJUMCTGA2DwAAxPAui0RwMAUHgrA0BQ8gGCEH6GNIlbIYgc7DMe6ZJtkGZapWEClGwBgBGkCRgBQZIqP6RjSFpubSaWpYJtB-S6nglYcBQjS5io5xQERMCVmAsZCSARnGaWrkQEJ6YmcMZkJNgnkmUFuoOU5LkQHihmXsZkCxf5KgJoFATBZ5qXJsQ0lZdkh7PIK0hEGemocNIKjACo16MP42DXioBw1BwswqAA5As1CaG1aalXUFVVXiAKkKG-UgrV-DXgAov6BBKo1zWtW1dooQ6PVZBq-WVfEHAhAA8tgACy7BtqCTVGC1eDtSt+LAlQ61pjM2DIBVngqFm1WIEY2j2Gmj3yC9KgwIgIQXaczgfRgE5VdIFTPbUHB4Gg7WwxAz0FogAAa2g2S+ACk9htY1jDQztaPIAjSMo3DyAY9jZKE8TpOo-DdRU21LO03GWPaAz1WsNI3M4+0ABsZI0PAjNHMzNOU8jHM03T2iNPjjgtYLqTC2LEtSyT2Cev9z0Vd4MoQ5VE5UJwcbCsjACsL5knrpYDhUpvIwAktgvpZKEesHH9tgAwNGZubViWQk1H14kGG1C54tXA6DFQEMgtUm8KyC-Qe2CjCoQZuLQiTSPoFWlbNNjSM4pQQNqUA1Pw0oZ2mRgNzK-g3DHZ5C4LHBINICcg2DKdp0KAOOMAE5PYDdDIMNo3lSC73t4gc9lakVDaD972HioAASAAqh1FO7GFIDNtAV60bRtWjXepETZh74fx+n4g59zdIDHqwLIRgIgWelinhVW+GFUhL2vD9AOgCg4h18kvLeWYd4T2SnQEw2gQHc3DIYcg494pdljqkLBqlEAMTcpA5K-t0yUMnjA8OaQPoIKQfFVBOgMGEOINg-+5tkr4LviQjhxCGIxRuOQrylDSzUPTLDfaR0Trr20CNKg3Ms4ThbrYNuiiha4OSkAlQb0o71S+u-Cu2ge59wAcZFh2gACEngLFeWsNITg2d4psP4ZwpenhJhuIUYY76ich4oHsCPU2FjJFXkQE4jg5hkFeRnqvMaVAnbGTcck7yod4p0OSZQyh9IDgHCAA)

Of course, this is only the most basic demo, and it may be necessary to export more control functions and states.

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by tolking**
### Description

close #2077

I wrote some [simple examples](https://tolking.github.io/animate-demo/), [source code](https://github.com/tolking/animate-demo)

### Additional context

I refer to the [control option](https://vueuse.org/guidelines.html#controls) to add `reactive` option for the useAnimate function to return different structures.

- The control and reactive have different semantics while return different structures.
- I also thought about splitting into two functions, but I did not find anything similar in this project.

---

### What is the purpose of this pull request? 

- [ ] Bug fix
- [X] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/core/useAnimate/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    })
+
+    expect(wrapper.vm.isSupported).toBe(false)
+    expect(wrapper.vm).not.toHaveProperty('pending')
+    expect(wrapper.vm).not.toHaveProperty('playState')
+    expect(wrapper.vm).not.toHaveProperty('replaceState')
+    expect(wrapper.vm).not.toHaveProperty('startTime')
+    expect(wrapper.vm).not.toHaveProperty('currentTime')
+    expect(wrapper.vm).not.toHaveProperty('timeline')
+    expect(wrapper.vm).not.toHaveProperty('playbackRate')
```
Comment (songjianet) [inline packages/core/useAnimate/index.test.ts:28]: At the end, it's better to use `wrapper.unmount()`

File: packages/core/useAnimate/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    })
+
+    expect(wrapper.vm.isSupported).toBe(false)
+    expect(wrapper.vm.pending).toBeDefined()
+    expect(wrapper.vm.playState).toBeDefined()
+    expect(wrapper.vm.replaceState).toBeDefined()
+    expect(wrapper.vm.startTime).toBeDefined()
+    expect(wrapper.vm.currentTime).toBeDefined()
+    expect(wrapper.vm.timeline).toBeDefined()
+    expect(wrapper.vm.playbackRate).toBeDefined()
```
Comment (songjianet) [inline packages/core/useAnimate/index.test.ts:49]: At the end, it's better to use `wrapper.unmount()`

File: packages/core/useAnimate/index.ts [hunk 3]
```diff
@@ -0,0 +1,330 @@
+import type { ComputedRef, Ref, ShallowRef, WritableComputedRef } from 'vue-demi'
```
Comment (songjianet) [inline packages/core/useAnimate/index.ts:1]: import type is best placed at the bottom of all references.

**Review by songjianet — CHANGES_REQUESTED** [review]
Can be used as a reference.

File: packages/core/useAnimate/index.ts (hunk 3 above)
Comment (tolking) [reply packages/core/useAnimate/index.ts:1]: Affected by formatting, this effect cannot be achieved. I have looked at other code formats and many places use this.

**Review by antfu — APPROVED** [review]

**Comment by Mrlilili** [comment]
Some tips for using this method: 在某些浏览器上会出现

**Pull request merged**